packages/components/src/components/domainproperties/dataset/DatasetDesignerPanels.tsx

```diff
@@ -232,44 +289,94 @@ export class DatasetDesignerPanelImpl extends React.PureComponent<Props & Inject
         const { setSubmitting } = this.props;
         const { model, fileImportData } = this.state;
 
-        saveDomain(model.domain, model.getDomainKind(), model.getOptions(), model.name)
-            .then(response => {
-                this.setState(
-                    produce((draft: Draft<State>) => {
-                        const updatedModel = draft.model;
-                        updatedModel.exception = undefined;
-                        updatedModel.domain = response;
-                    }),
-                    () => {
-                        // If we're importing Dataset file data, import file contents
-                        if (fileImportData) {
-                            this.handleFileImport();
-                        } else {
-                            const { model } = this.state;
+        let keyPropIndex;
+        let visitPropIndex;
+
+        if (this._participantId && this._sequenceNum) {
+            // filter out these fields
+            const updatedDomain = model.domain.merge({
+                fields: model.domain.fields
+                    .filter(field => field.name != this._participantId)
+                    .filter(field => field.name != this._sequenceNum)
```

Comment:
Instead of doing this as two separate filter loops over the fields, you could just combine them like:
```suggestion
                    .filter(field => field.name != this._participantId && field.name != this._sequenceNum)
```

Reply:
changed